src/solvers/dgsem_tree/dg_1d_parabolic.jl

```diff
@@ -530,18 +531,18 @@ function create_cache_parabolic(mesh::TreeMesh{1},
     elements = init_elements(leaf_cell_ids, mesh, equations_hyperbolic, dg.basis, RealT,
                              uEltype)
 
+    # Additions for parabolic
     n_vars = nvariables(equations_hyperbolic)
     n_nodes = nnodes(elements)
     n_elements = nelements(elements)
-    u_transformed = Array{uEltype}(undef, n_vars, n_nodes, n_elements)
-    gradients = similar(u_transformed)
-    flux_viscous = similar(u_transformed)
+
+    cache_viscous = CacheViscous1D{uEltype}(n_vars, n_nodes, n_elements)
```

Comment:
Related to the discussion above, maybe it would be nice to have an `init_viscous_data` function (for use here) and an `init_viscous_data!` function (for use in the AMR routines) that will return a `ViscousDataContainer`?

Reply:
The current version has an `init_viscous_container` function to be somewhat consistent with the existing containers, although this is just a wrapper for the constructor of `ViscousContainer1D`:

```julia
function init_viscous_container(n_vars::Integer, n_nodes::Integer,
                                n_elements::Integer,
                                ::Type{uEltype}) where {uEltype <: Real}
    return ViscousContainer1D{uEltype}(n_vars, n_nodes, n_elements)
end
```

Instead of providing something like `init_viscous_container!` I decided to implement a `resize!` function:

```julia
function Base.resize!(viscous_container::ViscousContainer1D, equations, dg, cache)
    capacity = nvariables(equations) * nnodes(dg) * nelements(dg, cache)
    resize!(viscous_container._u_transformed, capacity)
    resize!(viscous_container._gradients, capacity)
    resize!(viscous_container._flux_viscous, capacity)

    viscous_container.u_transformed = unsafe_wrap(Array,
                                                  pointer(viscous_container._u_transformed),
                                                  (nvariables(equations),
                                                   nnodes(dg),
                                                   nelements(dg, cache)))

    viscous_container.gradients = unsafe_wrap(Array,
                                              pointer(viscous_container._gradients),
                                              (nvariables(equations),
                                               nnodes(dg),
                                               nelements(dg, cache)))

    viscous_container.flux_viscous = unsafe_wrap(Array,
                                                 pointer(viscous_container._flux_viscous),
                                                 (nvariables(equations),
                                                  nnodes(dg),
                                                  nelements(dg, cache)))

    return nothing
end
```

Reply:
Sounds good, thanks!